plugin/internal/remotecall/remotecall.go

```diff
@@ -0,0 +1,235 @@
+
+	mappingKeyFile []byte
+
+	httpClient *http.Client
+}
+
+func InitRemoteCall(
+	userClusterNameURL string,
+	userClusterToken string,
+	clusterNameInClusterAuthenticatorURL map[string]string,
+	serviceJwtToken map[string]string,
+	clusterNameUpstreamURL map[string]string,
+	mappingKeyPath string,
+) (*RemoteCall, error) {
+	userClusterNameURLParsed, err := url.Parse(userClusterNameURL)
+	if err != nil {
+		return nil, fmt.Errorf("error parsing userClusterNameURL %q: %w", userClusterNameURL, err)
+	}
+
+	clusterNameInClusterAuthenticatorURLParsed := make(map[string]*url.URL, len(clusterNameInClusterAuthenticatorURL))
+
+	for clusterName, clusterURL := range clusterNameInClusterAuthenticatorURL {
+		clusterURLParsed, err := url.Parse(clusterURL)
+		if err != nil {
+			return nil, fmt.Errorf("error parsing URL %q: %w", clusterURL, err)
+		}
+		clusterNameInClusterAuthenticatorURLParsed[clusterName] = clusterURLParsed
+	}
+
+	jwtProviders := make(map[string]*ServiceJWTProvider, len(serviceJwtToken))
+	for clusterName, secret := range serviceJwtToken {
+		serviceJWTProvider, err := NewServiceJWTProvider(IdentitySSHGateway, []byte(secret), JwtValidity)
+		if err != nil {
+			return nil, fmt.Errorf("error creating jwt provider: %w", err)
+		}
+		jwtProviders[clusterName] = serviceJWTProvider
+	}
+
+	key, err := os.ReadFile(mappingKeyPath)
+	if err != nil {
+		return nil, fmt.Errorf("error reading mapping key: %w", err)
+	}
+
+	return &RemoteCall{
+		userClusterNameURL:                   userClusterNameURLParsed,
+		userClusterToken:                     userClusterToken,
+		clusterNameInClusterAuthenticatorURL: clusterNameInClusterAuthenticatorURLParsed,
+		serviceJwtProvider:                   jwtProviders,
+		httpClient:                           createHttpClient(),
+		mappingKeyFile:                       key,
+		clusterNameUpstreamURL:               clusterNameUpstreamURL,
+	}, nil
+}
+
+func createHttpClient() *http.Client {
+	return &http.Client{
+		Timeout: 10 * time.Second,
+		Transport: &http.Transport{
+			MaxIdleConns:    10,
+			IdleConnTimeout: 30 * time.Second,
+		},
+	}
+}
+
+func (r *RemoteCall) GetClusterName(username string) (string, error) {
+	if username == "" {
+		return "", fmt.Errorf("empty username")
+	}
+	req, err := http.NewRequest("GET", r.userClusterNameURL.JoinPath(username).String(), nil)
+	if err != nil {
+		return "", fmt.Errorf("error creating request for getting cluster name with url: %q and username:"+
+			" %q :%w", r.userClusterNameURL, username, err)
+	}
+
+	// Set custom headers if needed
+	req.Header.Set(UserAgent, UserAgentSSHGateway)
+	req.Header.Set(Accept, ApplicationJson)
+	req.Header.Set(ContentType, ApplicationJson)
+	req.Header.Set(AuthTokenUserClusterMapping, r.userClusterToken)
+
+	resp, err := r.httpClient.Do(req)
+	if err != nil {
+		return "", fmt.Errorf("error making request: %w", err)
+	}
+	if resp != nil && resp.Body != nil {
+		defer func() { _ = resp.Body.Close() }()
+	}
```

Comment:
there is no need for this you can just defer resp.Body.Close() because if err is nil resp will always be != nil